pkg/instrumentation/apachehttpd.go

```diff
@@ -0,0 +1,236 @@
+					// Create agent configuration file by pasting content of env var to a file
+					"echo \"$" + apacheAttributesEnvVar + "\" > " + apacheAgentConfDirFull + "/" + apacheAgentConfigFile + " && " +
+					"sed -i 's/" + apacheServiceInstanceId + "/'${" + apacheServiceInstanceIdEnvVar + "}'/g' " + apacheAgentConfDirFull + "/" + apacheAgentConfigFile + " && " +
+					// Include a link to include Apache agent configuration file into httpd.conf
+					"echo 'Include " + apacheConfigDirectory + "/" + apacheAgentConfigFile + "' >> " + apacheAgentConfDirFull + "/" + apacheConfigFile,
+			},
+			Env: []corev1.EnvVar{
+				{
+					Name:  apacheAttributesEnvVar,
+					Value: getApacheOtelConfig(pod, apacheSpec, index, otlpEndpoint, resourceMap),
+				},
+				{Name: apacheServiceInstanceIdEnvVar,
+					ValueFrom: &corev1.EnvVarSource{
+						FieldRef: &corev1.ObjectFieldSelector{
+							FieldPath: "metadata.name",
+						},
+					},
+				},
+			},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      apacheAgentVolume,
+					MountPath: apacheAgentDirFull,
+				},
+				{
+					Name:      apacheAgentConfigVolume,
+					MountPath: apacheAgentConfDirFull,
+				},
+			},
+		})
+	}
+
+	return pod
+}
+
+// Calculate if we already inject InitContainers.
+func isApacheInitContainerMissing(pod corev1.Pod, containerName string) bool {
+	for _, initContainer := range pod.Spec.InitContainers {
+		if initContainer.Name == containerName {
+			return false
+		}
+	}
+	return true
+}
+
+// Calculate Apache HTTPD agent configuration file based on attributes provided by the injection rules
+// and by the pod values.
+func getApacheOtelConfig(pod corev1.Pod, apacheSpec v1alpha1.ApacheHttpd, index int, otelEndpoint string, resourceMap map[string]string) string {
+	template := `
+#Load the Otel Webserver SDK
+LoadFile %[1]s/sdk_lib/lib/libopentelemetry_common.so
+LoadFile %[1]s/sdk_lib/lib/libopentelemetry_resources.so
+LoadFile %[1]s/sdk_lib/lib/libopentelemetry_trace.so
+LoadFile %[1]s/sdk_lib/lib/libopentelemetry_otlp_recordable.so
+LoadFile %[1]s/sdk_lib/lib/libopentelemetry_exporter_ostream_span.so
+LoadFile %[1]s/sdk_lib/lib/libopentelemetry_exporter_otlp_grpc.so
+#Load the Otel ApacheModule SDK
+LoadFile %[1]s/sdk_lib/lib/libopentelemetry_webserver_sdk.so
+#Load the Apache Module. In this example for Apache 2.4
+#LoadModule otel_apache_module %[1]s/WebServerModule/Apache/libmod_apache_otel.so
+#Load the Apache Module. In this example for Apache 2.2
+#LoadModule otel_apache_module %[1]s/WebServerModule/Apache/libmod_apache_otel22.so
+LoadModule otel_apache_module %[1]s/WebServerModule/Apache/libmod_apache_otel%[2]s.so
+#Attributes
+`
+	if otelEndpoint == "" {
+		otelEndpoint = "http://localhost:4317/"
+	}
+	serviceName := chooseServiceName(pod, resourceMap, index)
+	serviceNamespace := pod.GetNamespace()
+	// Namespace name override TBD
+
+	versionSuffix := ""
+	if apacheSpec.Version == "2.2" {
+		versionSuffix = "22"
+	}
```

Comment:
How was this version decided? it looks like there's also apache 2.4 above?

Reply:
explained in the comment in the code. 2.4 is default and the OTEL module file does not have any version suffix. Apache 2.2 has version suffix "22".